Sensor: BMI160: support interrupt for bmi160 i2c

lixuzha opened the pull request:
**sensor: bmi160: fix the issue of missing gpio cfg for i2c**

```
Add BMI160_TRIGGER_CFG in bmi160_cfg for i2c instance when
trigger mode enabled.
```

**sensor: bmi160: fix the issue with interrupt status**

```
When enabling trigger mode of bmi160 on i2c bus, I've come across
an issue where the readings register STATUS. The issue comes from
the dummy byte in the beginning of the buf.raw which is not necessary.
In addition, for bmi160 on spi bus, the dummy byte has already been
handled in bmi160_transceive while reading register.
So the dummy byte should be removed.
```

lixuzha committed:
sensor: bmi160: fix the issue with interrupt status

When enabling trigger mode of bmi160 on i2c bus, I've come across
an issue where the readings register STATUS. The issue comes from
the dummy byte in the beginning of the buf.raw which is not necessary.
In addition, for bmi160 on spi bus, the dummy byte has already been
handled in bmi160_transceive while reading register.
So the dummy byte should be removed.

Signed-off-by: Zhang Lixu <lixu.zhang@intel.com>

lixuzha committed:
sensor: bmi160: fix the issue of missing gpio cfg for i2c

Add BMI160_TRIGGER_CFG in bmi160_cfg for i2c instance when
trigger mode enabled.

Signed-off-by: Zhang Lixu <lixu.zhang@intel.com>

MaureenHelm commented:
Please take a look at the twister failure
```
FAILED: zephyr/drivers/sensor/bmi160/CMakeFiles/drivers__sensor__bmi160.dir/bmi160.c.obj 
ccache /usr/lib/llvm-15/bin/clang -DKERNEL -DTC_RUNID=8e9dda4baa2bd9c5434424427a0e8d36 -D_POSIX_C_SOURCE=200809 -D_XOPEN_SOURCE=600 -D_XOPEN_SOURCE_EXTENDED -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I../../../../../../include -Izephyr/include/generated -I../../../../../../soc/posix/inf_clock -I../../../../../../boards/posix/native_posix -I../../../../../../drivers/sensor/bmi160 -I../../../../../../subsys/testsuite/include -I../../../../../../subsys/testsuite/ztest/include -fno-strict-aliasing -Os -imacros /__w/zephyr/zephyr/twister-out/native_posix/tests/subsys/emul/subsys.emul/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fcolor-diagnostics -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-unused-but-set-variable -Wno-typedef-redefinition -Wno-deprecated-non-prototype -Wno-pointer-sign -Wpointer-arith -Wno-sometimes-uninitialized -Wno-shift-overflow -Wno-missing-braces -Wno-self-assign -Wno-address-of-packed-member -Wno-unused-function -Wno-initializer-overrides -Wno-section -Wno-unknown-warning-option -Wno-unused-variable -Wno-format-invalid-specifier -Wno-gnu -Wno-tautological-compare -Werror=implicit-int -fno-pic -fno-pie -Werror -fno-asynchronous-unwind-tables -Wno-vla -fmacro-prefix-map=/__w/zephyr/zephyr/tests/subsys/emul=CMAKE_SOURCE_DIR -fmacro-prefix-map=/__w/zephyr/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/__w/zephyr=WEST_TOPDIR -ffunction-sections -fdata-sections -m32 -include /__w/zephyr/zephyr/arch/posix/include/posix_cheats.h -std=c11 -MD -MT zephyr/drivers/sensor/bmi160/CMakeFiles/drivers__sensor__bmi160.dir/bmi160.c.obj -MF zephyr/drivers/sensor/bmi160/CMakeFiles/drivers__sensor__bmi160.dir/bmi160.c.obj.d -o zephyr/drivers/sensor/bmi160/CMakeFiles/drivers__sensor__bmi160.dir/bmi160.c.obj -c /__w/zephyr/zephyr/drivers/sensor/bmi160/bmi160.c
/__w/zephyr/zephyr/drivers/sensor/bmi160/bmi160.c:1023:29: error: use of undeclared identifier '__device_dts_ord_DT_N_S_i2c_100_S_bmi_68_P_int_gpios_IDX_0_PH_ORD'
DT_INST_FOREACH_STATUS_OKAY(BMI160_DEFINE)
                            ^
/__w/zephyr/zephyr/drivers/sensor/bmi160/bmi160.c:1023:29: error: use of undeclared identifier 'DT_N_S_i2c_100_S_bmi_68_P_int_gpios_IDX_0_VAL_pin'
2 errors generated.
```

lixuzha committed:
tests: build_all: add int-gpios in test_i2c_bmi160 to fix build issue

After adding the interrupt support for bmi160 i2c instance, it needs to add
the int-gpios in the twister test's dts because the bmi160's trigger mode
is enabled by default.

Signed-off-by: Zhang Lixu <lixu.zhang@intel.com>

lixuzha committed:
tests: emul: disable bmi160 trigger mode in emul's prj.conf

The trigger mode is enabled by default in bmi160's Kconfig.
After adding the interrupt support for bmi160 i2c instance,
it needs to disable the trigger mode if there is not int-gpios
in dts.

Signed-off-by: Zhang Lixu <lixu.zhang@intel.com>

lixuzha commented:
The twister failure is fixed now. Thanks.
